### PR TITLE
Multiple Includes/Excludes

### LINKED ISSUE #474: [Frontend][Publisher] Update definitions for multiple includes/excludes

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Updates Metric Configuration's (and other affected components') includes/excludes to conform to the new API and data structure. 

Note: this PR's diff is larger than it really is as the largest diff was updating the GuidanceStore tests which used the old data structure to access metric/dimension includes/excludes.

Playtesting Demo: [mahmoudtest---justice-counts-web-qqec6jbn6a-uc.a.run.app](https://mahmoudtest---justice-counts-web-qqec6jbn6a-uc.a.run.app/)

Demo:
 
https://user-images.githubusercontent.com/59492998/223876307-83d8f48c-a82a-416c-876d-1659c75896c6.mov

## Related issues

Closes #474 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by mxosman** [comment]
I did a quick test of this and haven't seen any issues - things are saving as expected. Would love a sanity check though!

I still need to test the Guidance functions and the LearnMoreModal in Agency Dashboards that were affected by this.

**Comment by mxosman** [comment]
What are your thoughts on the design? I went with the usual minimal grey text and a faint line separator to separate the categories. Open to thoughts and experimentation!

File: agency-dashboard/src/DashboardModals/LearnMoreModal.tsx [hunk 1]
```diff
@@ -70,15 +70,17 @@ export const LearnMoreModal: React.FC<{
...
             <>
               <ModalSectionTitle>Settings</ModalSectionTitle>
-              {metric.settings.map((setting) => (
-                <ModalParagraph>
-                  <span>{setting.label}</span>
-                  {`: ${setting.included}`}
-                </ModalParagraph>
-              ))}
+              {metric.includes_excludes?.forEach((includesExcludes) =>
+                includesExcludes.settings.map((setting) => (
```
Comment (lilidworkin) [inline agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Should we include the include exclude description as a header here too?

**Review by lilidworkin — COMMENTED** [review]
This looks good to me! I'm going to run and test it out locally. Just to make sure I remember/understand correctly, the POST flow didn't really change much here, is that right? It's the GET flow that is now consuming the data in a different way?

**Comment by lilidworkin** [comment]
Tested and looks great to me! I like the design!

File: agency-dashboard/src/DashboardModals/LearnMoreModal.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Good call! I haven't tested this yet - but I think so!

File: publisher/src/components/MetricConfiguration/MetricBreakdownAvailabilityDefinitions.tsx [hunk 2]
```diff
@@ -147,40 +148,69 @@ export const MetricBreakdownAvailabilityDefinitions: React.FC<MetricDefinitionsP
     const revertToAndSaveDefaultValues = () => {
       if (systemSearchParam && metricSearchParam) {
         /** Create array of default settings and update settings (to default value) in the store */
-        const defaultSettings = activeSettingsKeys.map((settingKey) => {
-          let currentSettingDefaultValue;
+        const defaultSettings: MetricSettings["settings"] = [];
```
Comment (morden35) [inline publisher/src/components/MetricConfiguration/MetricBreakdownAvailabilityDefinitions.tsx:151]: should this be MetricSettings["includes_excludes"]?

File: agency-dashboard/src/DashboardModals/LearnMoreModal.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Unrelated to this - but when I ran Agency Dashboard - it's crashing when I click on the Learn More link - specifically here:

TypeError: Cannot read properties of undefined
```
              {metric.definitions.map((definition) => (
                <ModalParagraph key={definition.term}>
                  <span>{definition.term}</span>
                  {`: ${definition.definition}`}
                </ModalParagraph>
              ))}
```

I noticed there is no `definitions` property in the backend response. Do we no longer have metric definitions? If not, I can remove this portion of the code in the LearnMoreModal.

**Comment by mxosman** [comment]
> I'm loving this @mxosman! So cool to see.
> 
> I personally like this design! 2 tiny nits would be maybe making the description text slightly darker? or a different color? I find it a little hard to read the light grey. I would also make the separating line either a little thicker or darker (just more pronounced in general)

What do you think of this? Let me know if you would boost it even more!

Reference:
Screenshot 2023-03-09 at 2 42 15 PM


Darker:
Screenshot 2023-03-09 at 2 42 31 PM

File: agency-dashboard/src/DashboardModals/LearnMoreModal.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Actually, I could conditionally render this portion based on whether or not there is a `metrics.definitions`. That fixes this - but still curious about the definitions property not being present in the response.

Comment (morden35) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: definition was deprecated [here](https://github.com/Recidiviz/recidiviz-data/pull/19022/files)

**Comment by morden35** [comment]
> > I'm loving this @mxosman! So cool to see.
> > I personally like this design! 2 tiny nits would be maybe making the description text slightly darker? or a different color? I find it a little hard to read the light grey. I would also make the separating line either a little thicker or darker (just more pronounced in general)
> 
> What do you think of this? Let me know if you would boost it even more!
> 
> Reference: Screenshot 2023-03-09 at 2 42 15 PM
> 
> Darker: Screenshot 2023-03-09 at 2 42 31 PM

I like it! Thank you!

File: agency-dashboard/src/DashboardModals/LearnMoreModal.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Thank you, Michelle - completely missed that!

Comment (morden35) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx:77]: Totally on us! We should have checked if this broke to front end before merging

Comment (mxosman) [reply agency-dashboard/src/DashboardModals/LearnMoreModal.tsx]: Not at all! I should be looking more closely at those PRs too. I just haven't ran agency dashboards in a while - I'll need to do a more thorough exploration there to see if anything else needs an update!

**Comment by mxosman** [comment]
Gave the Learn More modal in Agency Dashboards similar treatment:

Screenshot 2023-03-09 at 3 41 53 PM

**Pull request merged**